### PR TITLE
feat(retention): add multi breakdown frontend filter support, behind flag

### PR DESCRIPTION
**Pull request by hazzadous**
## Changes

This change is build on top of https://github.com/PostHog/posthog/pull/7431 
and simply adds the react breakdown component to the retention insight tab.

I had to refactor a little bit to make the component not dependent on knowledge of the caller
by passing in if we want to use multiple breakdowns. It's the caller that will 
know this best, not the component., as the blocker here is if the API for that 
specific insight call supports it.

Screenshot 2021-12-01 at 15 08 19

## How did you test this code?

manually via the storybook

### CODE REVIEW

File: frontend/src/scenes/insights/BreakdownFilter/TaxonomicBreakdownFilter.tsx [hunk 1]
```diff
@@ -103,7 +99,7 @@ export function BreakdownFilter({ filters, setFilters, buttonType }: TaxonomicBr
 
         if (changedBreakdownType) {
             let newFilters: Partial<FilterType>
-            if (multiPropertyBreakdownIsEnabled) {
+            if (useMultiBreakdown) {
```
Comment (pauldambra) [inline frontend/src/scenes/insights/BreakdownFilter/TaxonomicBreakdownFilter.tsx:102]: I think there are changes in master that will clash here

**Review by pauldambra — APPROVED** [review]
love that you have added a story book! LGTM

File: frontend/src/scenes/insights/BreakdownFilter/TaxonomicBreakdownFilter.tsx (hunk 1 above)
Comment (hazzadous) [reply frontend/src/scenes/insights/BreakdownFilter/TaxonomicBreakdownFilter.tsx:102]: ah cool I'll rebase

Comment (hazzadous) [reply frontend/src/scenes/insights/BreakdownFilter/TaxonomicBreakdownFilter.tsx:102]: 💥

**Comment by hazzadous** [comment]
> love that you have added a story book

@pauldambra I find it useful to be able to isolate changes, I hope others do 🙏

**Pull request merged**